pulp_python/app/models.py

```diff
@@ -178,6 +182,7 @@ class PythonPackageContent(Content):
     classifiers = models.JSONField(default=list)
     project_urls = models.JSONField(default=dict)
     description_content_type = models.TextField()
+    dynamic = models.TextField()
```

Comment:
I think we are missing `provides_extra`. Added in 2.1. Also, can we reformat the list to be like pkginfo? https://github.com/jwilk-mirrors/python-pkginfo/blob/master/pkginfo/distribution.py#L124

Reply:
Sure. For `provides_extra`, are we going to be missing metadata for existing packages?  I don't think we ever had a problem with rejection of metadata 2.1 packages as we did with 2.3

How do you want to deal with that?

Reply:
Also, I filed https://github.com/pypi/warehouse/issues/16190 because I'm having difficulty figuring out exactly what these fields are meant to look like.  The core metadata specification describes how they're represented in the metadata files within a package, not how they're represented on PyPI.

Reply:
> Sure. For `provides_extra`, are we going to be missing metadata for existing packages? I don't think we ever had a problem with rejection of metadata 2.1 packages as we did with 2.3
> 
> How do you want to deal with that?

We might need to create a new endpoint that runs a task to fix bad metadata. Unless you are uploading every package most python content will have incorrect metadata because syncing doesn't individually check the metadata for each package. Currently we take the latest's package metadata and re-use it for every version of that package.